src/__tests__/fireEvent.test.js

```diff
@@ -163,3 +164,63 @@ test('event with multiple handler parameters', () => {
 
   expect(handlePress).toHaveBeenCalledWith('param1', 'param2');
 });
+
+test('should not fire on disabled TouchableOpacity', () => {
+  const handlePress = jest.fn();
+  const screen = render(
+    <TouchableOpacity onPress={handlePress} disabled={true}>
+      <Text>Trigger</Text>
+    </TouchableOpacity>
+  );
+
+  expect(() => fireEvent.press(screen.getByText('Trigger'))).toThrow(
+    'No handler function found for event: "press"'
+  );
+  expect(handlePress).not.toHaveBeenCalled();
+});
+
+test('should not fire on disabled Pressable', () => {
+  const handlePress = jest.fn();
+  const screen = render(
+    <Pressable onPress={handlePress} disabled={true}>
+      <Text>Trigger</Text>
+    </Pressable>
+  );
+
+  expect(() => fireEvent.press(screen.getByText('Trigger'))).toThrow(
+    'No handler function found for event: "press"'
+  );
+  expect(handlePress).not.toHaveBeenCalled();
+});
+
+test('should pass event up on disabled TouchableOpacity', () => {
+  const handleInnerPress = jest.fn();
+  const handleOuterPress = jest.fn();
+  const screen = render(
+    <TouchableOpacity onPress={handleOuterPress}>
+      <TouchableOpacity onPress={handleInnerPress} disabled={true}>
+        <Text>Inner Trigger</Text>
+      </TouchableOpacity>
+    </TouchableOpacity>
+  );
+
+  fireEvent.press(screen.getByText('Inner Trigger'));
+  expect(handleInnerPress).not.toHaveBeenCalled();
+  expect(handleOuterPress).toHaveBeenCalledTimes(1);
+});
+
+test('should pass event up on disabled Pressable', () => {
```

Comment:
Can we add one more test with a component that has "disabled" prop but doesn't pass it to the native View?